### PR TITLE
Fix/terms block taxonomies no terms

### PR DESCRIPTION
**Pull request by dpellenwood**
## What does this do/fix?

* Adds a no terms message in the block editor (based on Chris's REST_REQUEST approach). 
* Simplifies the render to use the same markup for all contexts where there are terms (both single & multiple)

### CODE REVIEW

File: wp-content/themes/core/blocks/tribe/terms/render.php [hunk 1]
```diff
@@ -15,45 +15,35 @@
 
 echo '<div ' .  wp_kses_data( get_block_wrapper_attributes() ) . '>';
 
-if ( 0 === count( $terms_block_terms ) ) {
-	echo '<!-- Terms block: No terms to list. -->';
-	echo '</div>';
-
-	return;
+// No terms and we're in the block editor
+if ( 0 === count( $terms_block_terms ) && ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
```
Comment (dpellenwood) [inline wp-content/themes/core/blocks/tribe/terms/render.php:19]: I think we should probably extract `( defined( 'REST_REQUEST' ) && REST_REQUEST ) )` into a utility/helper somewhere more global. It feels weird to check like this in the context. But, food for thought.

File: wp-content/themes/core/blocks/tribe/terms/render.php [hunk 2]
```diff
@@ -15,45 +15,35 @@
...
 }
 
-if ( count( $terms_block_terms ) > 1 ) {
-	echo  '<ul class="wp-block-tribe-terms__list">';
-}
+if ( count( $terms_block_terms ) > 0 ) {
+	echo '<ul class="wp-block-tribe-terms__list">';
 
-foreach ( $terms_block_terms as $term ) {
-	if ( count( $terms_block_terms ) > 1 ) {
```
Comment (dpellenwood) [inline wp-content/themes/core/blocks/tribe/terms/render.php:30]: I don't think it's worth switching the markup depending on if there is only a single term displayed. The block itself is a "terms block" which infers there could be one or more.

**Review by dpellenwood — COMMENTED** [review]
Left a couple notes on my own code for conversation.

File: wp-content/themes/core/blocks/tribe/terms/render.php [hunk 3]
```diff
@@ -15,45 +15,35 @@
 
 echo '<div ' .  wp_kses_data( get_block_wrapper_attributes() ) . '>';
 
-if ( 0 === count( $terms_block_terms ) ) {
-	echo '<!-- Terms block: No terms to list. -->';
```
Comment (dpellenwood) [inline wp-content/themes/core/blocks/tribe/terms/render.php:19]: We render the block wrapper div regardless, so I don't see the comment as necessary.

File: wp-content/themes/core/blocks/tribe/terms/style.pcss [hunk 4]
```diff
@@ -52,3 +52,14 @@
...
+/* -----------------------------------------------------------------------
+ * Terms Block - Empty Terms List
+ * ----------------------------------------------------------------------- */
+
+.wp-block-tribe-terms__empty-msg {
+
+	.editor-styles-wrapper & {
+		color: var(--color-neutral-60);
+	}
+}
```
Comment (GeoffDusome) [inline wp-content/themes/core/blocks/tribe/terms/style.pcss:65]: ❓ Should we add the `t-tag` mixin here so the styling shows correctly depending on the style chosen or is it ok that this displays differently?

**Review by GeoffDusome — APPROVED** [review]
🟢 This works for me - kinda bridges both solutions. Just one question.

File: wp-content/themes/core/blocks/tribe/terms/style.pcss (hunk 4 above)
Comment (dpellenwood) [reply wp-content/themes/core/blocks/tribe/terms/style.pcss]: Good catch. Updated.

**Comment by dpellenwood** [comment]
Made a few additional changes to reduce the style bleed, and align with how core renders the Post Tags block in the editor when no terms are assigned.

File: wp-content/themes/core/blocks/tribe/terms/editor.pcss [hunk 5]
```diff
@@ -1,16 +1,14 @@
-.wp-block-tribe-terms {
+.wp-block-tribe-terms {}
```
Comment (GeoffDusome) [inline wp-content/themes/core/blocks/tribe/terms/editor.pcss:1]: ❓ Do we need this empty selector?

**Review by GeoffDusome — APPROVED** [review]
🟢 Just one question, other than that good to go!

**Review by ChrisMKindred — APPROVED** [review]
👍

File: wp-content/themes/core/blocks/tribe/terms/editor.pcss (hunk 5 above)
Comment (dpellenwood) [reply wp-content/themes/core/blocks/tribe/terms/editor.pcss]: Good catch. Removed.

**Pull request merged**